apps/core/tasks/auto_upgrade/runner.py

```
@@ -0,0 +1,300 @@
+    """Return the local URL used to probe the Django suite."""
+
+    lock_dir = base_dir / ".locks"
+    mode_file = lock_dir / "nginx_mode.lck"
+    mode = "internal"
+    if mode_file.exists():
+        try:
+            value = mode_file.read_text(encoding="utf-8").strip()
+        except OSError:
+            value = ""
+        if value:
+            mode = value.lower()
```

Ruff (0.14.14)

[error] Local variable `mode` is assigned to but never used

Remove assignment to unused variable `mode`

(F841)